dev/AppNotifications/NotificationProgressData.cpp

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "AppNotificationProgressData.h"
+#include "NotificationProgressData.h"
+#include "AppNotificationUtility.h"
+
+namespace winrt
+{
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+}
+
+namespace ToastABI
+{
+    using namespace ::ABI::Microsoft::Internal::ToastNotifications;
+}
+
+NotificationProgressData::NotificationProgressData(winrt::AppNotificationProgressData const& progressData)
+    : m_progressData(progressData.SequenceNumber())
+{
+    m_progressData.Status(progressData.Status());
+    m_progressData.Title(progressData.Title());
+    m_progressData.Value(progressData.Value());
+    m_progressData.ValueStringOverride(progressData.ValueStringOverride());
+}
+
+STDMETHODIMP NotificationProgressData::get_SequenceNumber(_Out_ unsigned int* value) noexcept
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = m_progressData.SequenceNumber();
+
+    return S_OK;
+}
+
+STDMETHODIMP NotificationProgressData::get_Title(_Out_ HSTRING* value) noexcept try
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = Microsoft::Windows::AppNotifications::Helpers::safe_make_unique_hstring(m_progressData.Title().c_str()).release();
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP NotificationProgressData::get_Value(_Out_ double* value) noexcept
+{
+    auto lock{ m_lock.lock_shared() };
+
```

Comment:
Nit - are these locks required? There are no "update" methods here, only read. The data is set in the constructor and then effectively read-only throughout.

Reply:
@sharath2727

Reply:
As a principle, we are locking all read/write operations. But in this case since it's a one-time write for all member variables, we don't need to lock this class.

Reply:
Fixing.